f8a_report/snyk_token_validation_main.py

```diff
@@ -26,5 +30,20 @@ def call_snyk_api(user_to_tokens: dict) -> list:
     return unregistered_users
 
 
+def cache_all_users(users: dict):
+    """Cache users in PVC."""
+    for user in users:
+        user_cache = {
+            "user_id": user[0], "snyk_api_token": user[1],
+            "last_validated_date": user[2], "status": user[3],
+            "registered_date": user[4], "created_date": user[5],
+            "updated_date": user[6], "user_source": user[7]
+        }
+
+        # Create file for each user into PVC having details about user
+        with open(DB_CACHE_FILE_PATH + "/" + user[0] + ".json", 'w', encoding='utf-8') as file:
```

Comment:
can you check how much latency are we adding via this file open/close operation. wont affect here, but might affect the normal CA calls

Reply:
Okay, as this is a separate component it may take some time to complete the operation because it needs to recreate (it will delete old file for the user and create new with updated data) files daily for all the entries available in user table of RDS, but in SA/CA we will have to read only one file per request and file size is very small, so I guess it should be faster then existing RDS query time.